### DISCUSSION TITLE
var/tmp folder flooded with temp-file

### DISCUSSION THREAD

**Discussion by hirakch** [pimcore/pimcore#11738] [Q&A / Support]
In a certain day from nowhere the var/tmp folder is flooded with temporary files and in some cases 2 files with same size (seems to be duplicate for multiple thumbnails)

The maintenance job runs at the same time when these temporary files started creating, otherwise couldn't find anything. I can see there is a shutdown function registered in `lib/Helper/TemporaryFileHelperTrait.php`  to delete the temp file once thumbnail is generated, but why that didn't work? anyone faced similar issues?

<img width="737" alt="image" src="https://user-images.githubusercontent.com/10167719/159551319-94935b39-4b2b-40a4-8414-5eb7b48b2be0.png">

**Comment by NiklasBr** [comment]
We have the same question about the 60k files in our `var/tmp` folder as well. Growing daily.

**Comment by craigatkinson** [comment]
Interesting, we have 42k files in only 3 weeks.

**Comment by brusch** [comment]
@hirakch @NiklasBr @craigatkinson Please have a look at #11566 which was released with `10.3.2` - this should fixes the issue. 😊

**Reply by NiklasBr** [reply]
Running `10.3.3` and the issue is still here @brusch

**Reply by dvesh3** [reply]
maybe wait for this then https://github.com/pimcore/pimcore/pull/11862

**Reply by NiklasBr** [reply]
@dvesh3 still a problem with 2024.2 (Pimcore 11.3.2.1), the number of files is slowly growing and passed 14k items recently:

```shell
www-data@e9354e1f45f7:~/pimcore/var/tmp$ find . -name "temp-file-*" -type f  | wc -l
14468
```

It's image files like:
```shell
./temp-file-66154ddec5a1b-80a6e83d233af64ca3e33a370730ac.webp
./temp-file-6628448ea1fd3-3ac41604396f095706a3b27a5c020b.png
./temp-file-66497536e5840-f9dd29a377301105b1a4c59c0b1b8d.png
./temp-file-66f25bc3a3790-5ea08923ac9b42b04e840533d76867.jpg
./temp-file-667b7ccf8eab1-e1ad8865dc04147d316e4871fdf567.png
./temp-file-6618a318b0270-d36d210b41dbbb0536538b253086d5.jpg
```

**Reply by cancan101** [reply]
This PR might address the symptoms: https://github.com/pimcore/pimcore/pull/17560 though it isn't addressing the root cause (whatever that may be that makes the files).
